grpc.go

```diff
@@ -536,6 +541,91 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 	}, nil
 }
 
+func getPIDIndex(title string) int {
+	// looking for PID field in ps title
+	fields := strings.Fields(title)
+	for i, f := range fields {
+		if f == "PID" {
+			return i
+		}
+	}
+	return -1
+}
+
+func (a *agentGRPC) ListProcesses(ctx context.Context, req *pb.ListProcessesRequest) (*pb.ListProcessesResponse, error) {
+	resp := &pb.ListProcessesResponse{}
+
+	c, err := a.sandbox.getContainer(req.ContainerId)
+	if err != nil {
+		return resp, err
+	}
+
+	pids, err := c.container.Processes()
+	if err != nil {
+		return resp, err
+	}
+
+	switch req.Format {
+	case "table":
+	case "json":
+		resp.ProcessList, err = json.Marshal(pids)
+		return resp, err
+	default:
+		return resp, fmt.Errorf("invalid format option")
+	}
+
```

Comment:
@devimc everything after this looks like it is not reachable. Am I missing something ?

Reply:
do you mean if the format is not supported ?

Reply:
I mean that if you look at this `switch` statement, there are `return` for each statement, including `default`. How do you expect the code further down to be reached ?

Reply:
no,  `table` doesn't have a `return`

```go
......
	case "table":
	case "json":
		resp.ProcessList, err = json.Marshal(pids)
		return resp, err
.......
```

Reply:
Oh yes I was assuming a `fallthrough` here...

Reply:
@sboeuf - I think the code is validating `table` as a legitimate format type (but with no associated statement). However, the actual handling for `table` is below the `switch`.